File: Xamarin/Common/Model/Location.cs

using Newtonsoft.Json.Linq;

namespace PropertyFinder.Model
{
  public class Location
  {
    public Location(JToken jsonLocation)
    {
      DisplayName = (string)jsonLocation["long_title"];
      Name = (string)jsonLocation["place_name"];
    }

    public Location(string location)
    {
      DisplayName = location;
      Name = location;
    }

    public string DisplayName { private set; get; }

    public string Name { private set; get; }
  }
}

File: native/android/PropertyCross/src/com/propertycross/android/model/Location.java
package com.propertycross.android.model;

import org.json.JSONException;
import org.json.JSONObject;

public class Location {

	private String displayName;
	private String name;
	
	public Location(JSONObject jsonLocation) {
		try {
	      displayName = jsonLocation.getString("long_title");
	      name = jsonLocation.getString("place_name");
	    }
		catch (JSONException e) {
	      displayName = "";
	      name = "";
	    }
	}
	
	public Location(String location) {
		displayName = location;
		name = location;
	}
	
	public String getDisplayName() {
		return displayName;
	}
	
	public String getName() {
		return name;
	}
}
